### PR TITLE
feat: upload and view TMUA grades

### PR DESCRIPTION
**Pull request by zaki-amin**
**UI**
- `TmuaGradeBox` displays TMUA grades in a table on Admin and UG tutor forms (hidden from reviewers)
  - Screenshot 2024-10-02 at 14 22 11

**Server**
- TMUA Grade upload actually updates TMUA grades (was a no-op placeholder before)
  - Example upload file: `public/dev-example-tmua.csv`
- Make `CandidateCallout` component to avoid duplication and complement
- `utils.ts` for Prisma Decimal -> TS number conversion

### CODE REVIEW

File: components/DataUploadDialog.tsx [hunk 1]
```diff
@@ -91,6 +91,9 @@ const DataUploadDialog: FC<DataUploadDialogProps> = ({ disabled }) => {
       <FormWrapper action={processCsvUploadWrapped} submitButtonText="Upload">
         <DataUploadForm file={file} setFile={setFile} />
       </FormWrapper>
+      <Button color="teal" onClick={() => window.location.reload()}>
```
Comment (procaconsul) [inline components/DataUploadDialog.tsx:94]: This is interesting. I was going to suggest "Save", but that would be misleading for people who do _not_ press it, who may think that the data is _not_ loaded in that case.

**Review by procaconsul — APPROVED** [review]

File: components/DataUploadDialog.tsx (hunk 1 above)
Comment (zaki-amin) [reply components/DataUploadDialog.tsx:94]: It's a little difficult as a situation, we don't want to force refreshing of the page so the user sees the message about successful/failed insertions and so the user can continue uploading data. However, the table itself won't update to include any new applications if the user clicks off and the page isn't refreshed.

Comment (zaki-amin) [reply components/DataUploadDialog.tsx:94]: The styling of this is a bit off right now too but I plan on doing all the styling together later on.

Comment (procaconsul) [reply components/DataUploadDialog.tsx:94]: Fine by me, Sir

**Pull request merged**